src/components/PageMainLayout.tsx

```diff
@@ -0,0 +1,28 @@
+import { ReactNode } from 'react';
+import { styled } from 'styled-components';
+
+interface Props {
+  children: ReactNode;
```

Comment:
### JSX.Element

```tsx
interface JSX.Element

/**
 * @deprecated Use `React.JSX` instead of the global `JSX` namespace.
 */
interface Element extends React.ReactElement<any, any> { }
```

React 컴포넌트만 올 수 있습니다. 단일 자식 요소 타입으로 여러 개의 요소를 전달할 수 없습니다. 

### ReactNode

```tsx
type ReactNode = string | number | boolean | ReactElement<any, string | JSXElementConstructor<any>> | ReactFragment | ReactPortal | null | undefined
```

범용적인 타입입니다. prop으로 단일 요소 또는 여러 개의 요소를 전달할 수 있습니다.

### ReactChild

단일 React 요소를 나타내거나 문자열, 숫자 등의 원시 타입을 포함할 수 있습니다. `deprecated` 예정으로 안 쓰는 것이 좋을 것 같습니다. 

```tsx
*@deprecated* — - This type is not relevant when using React. 
Inline the type instead to make the intent clear.

type ReactChild = string | number | ReactElement<any, string | JSXElementConstructor<any>>
```

여러 요소를 전달하다보니 타입 에러가 발생해서 `ReactNode`를 사용했는데, 타입을 더 좁히기 위해서 `JSX.Element`를 사용하는 것이 좋을 것 같습니다!

Reply:
굿
좋은 공부하셨네요!